tests/splitting/test_split_temporal_filter.py

```diff
@@ -0,0 +1,34 @@
+# This file is part of LensKit.
+# Copyright (C) 2018-2023 Boise State University
+# Copyright (C) 2023-2025 Drexel University
+# Licensed under the MIT license, see LICENSE.md for details.
+# SPDX-License-Identifier: MIT
+
+import datetime as dt
+
+import numpy as np
+
+from pytest import approx
+
+from lenskit.data import Dataset
+from lenskit.splitting import split_global_time, split_temporal_fraction
+
+
+def test_temporal_split_filtered_users(ml_ds: Dataset):
+    point = dt.datetime.fromisoformat("2015-01-01")
+
+    split = split_global_time(ml_ds, "2015-01-01", filter_test_users=True)
+    assert np.all(split.train.interaction_table(format="pandas")["timestamp"] < point)
+
+    for u, il in split.test:
+        ts = il.field("timestamp", format="pandas")
+
+        assert ts is not None
+        assert np.all(ts >= point)
```

Comment:
we also need to verify that all of the test users have at least one training rating.